MRN-Code/coinstac
# Redesign/consortium pipeline pages

* **Problem:**
   1. The redesign requires new consortium and pipeline creation pages.
* **Solution:**
   1. Add consortia list page featuring all available consortia with options to join and leave.
   1. Add a tabbed `consortium` page with tabs for "About", "Pipeline" and "Results".
   1. Add a pipeline list page
   1. Add a pipeline creation page with drag/drop computations to form pipelines and mappable covariates that can grab input from previous computations' output.
* **Tests:**
   1. Follow the steps outlined in `packages/coinstac-api-server/README.md` to setup your development environment with RethinkDB, start the GraphQL API, and insert a couple computations into the new database.
   1. Duplicate `packages/coinstac-graphql/config/api-config-sample.json` as `api-config.json`.
   1. Uncomment the contents and fill in a random string for the `jwtSecret`.
   1. Duplicate `packages/coinstac-graphql/config/comp-reg-config-sample.json` as `comp-reg-config.json` and uncomment the contents.
   1. Start COINSTAC UI dev environment as usual. 
   1. Start the application and login with user: `test` & pass: `password` to test authentication.
   1. Visit the *Consortium* navigation item to see a list of test consortia.
   1. View details for listed consortia and navigate through tabs.
   1. Click on the *Pipelines* navigation item to see a list of pipelines (no test ones available currently).
   1. Click the _Create new pipeline_ button to see pipeline creation page.
   1. Add a couple computations to the pipeline using the labeled dropdown.
   1. Rearrange computations by dragging them around.
   1. Add a covariate with the type of number to see listed mapping options. Computations listed later in the list should have earlier computations output listed as possible inputs.
